test/integration/order.test.ts

```diff
@@ -0,0 +1,177 @@
+            message:
+              "Not match in '[0-9A-Fa-f]{8}-[0-9A-Fa-f]{4}-4[0-9A-Fa-f]{3}-[89ABab][0-9A-Fa-f]{3}-[0-9A-Fa-f]{12}'",
+            value: '789ad47',
+          },
+        },
+        name: 'ValidateError',
+        message: 'Validation failed',
+      })
+    })
+  })
+
+  describe('if invalid attachment uuid', () => {
+    beforeEach(async () => {
+      res = await post(app, '/order', { parametersAttachmentId: 'a789ad47' })
+    })
+
+    it('returns 422 along with validation error', () => {
+      const { status, body } = res
+
+      expect(status).to.equal(422)
+      expect(body).to.deep.contain({
+        fields: {
+          '.parametersAttachmentId': {
+            message:
+              "Not match in '[0-9A-Fa-f]{8}-[0-9A-Fa-f]{4}-4[0-9A-Fa-f]{3}-[89ABab][0-9A-Fa-f]{3}-[0-9A-Fa-f]{12}'",
+            value: 'a789ad47',
+          },
+        },
+        name: 'ValidateError',
+        message: 'Validation failed',
+      })
+    })
+  })
+
+  it('retrieves order by id', async () => {
+    const { status, body } = await post(app, '/order', { parametersAttachmentId })
+
+    expect(status).to.equal(201)
+    expect(body).to.have.property('id')
+    expect(body).to.deep.contain({
+      owner: 'test-self',
+      state: 'created',
+      parametersAttachmentId: 'a789ad47-91c3-446e-90f9-a7c9b233eaf8'
+    })
+  })
+
+  it('should create an order demand', async () => {
+    const response = await post(app, '/order', { parametersAttachmentId })
+    const { id: responseId, ...responseRest } = response.body
+
+    expect(response.status).to.equal(201)
+    expect(responseId).to.match(
+      /^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[1-5][0-9a-fA-F]{3}-[89ABab][0-9a-fA-F]{3}-[0-9a-fA-F]{12}$/
+    )
+    expect(responseRest).to.deep.equal({
+      parametersAttachmentId,
+      state: DemandState.created,
+      owner: selfAlias,
+    })
+  })
+
+  it('creates an order transaction on chain', async () => {
+    const { body: { id: orderId } } = await post(app, '/order', { parametersAttachmentId })
+
+    // submit to chain
+    const response = await post(app, `/order/${orderId}/creation`, {})
+    expect(response.status).to.equal(201)
+
+    const { id: transactionId, state } = response.body
+    expect(transactionId).to.match(
+      /^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[1-5][0-9a-fA-F]{3}-[89ABab][0-9a-fA-F]{3}-[0-9a-fA-F]{12}$/
+    )
+    expect(state).to.equal(TransactionState.submitted)
+
+    // check local transaction updates
+    const [transaction] = await db.getTransaction(transactionId)
+    expect(transaction.state).to.equal(TransactionState.finalised)
+    expect(transaction).to.contain({
+      state: 'finalised',
+      apiType: 'order',
+      transactionType: 'creation',
+    })
+
+    const [order] = await db.getDemand(orderId)
+    expect(transaction.localId).to.equal(orderId)
+    expect(order).to.contain({
+      id: orderId,
+      state: 'created',
+      subtype: 'order',
+      parametersAttachmentId,
+    })
```

Comment:
would be good to check token ids on the order too

Reply:
to show it's been put on chain and its token ID has updated locally

Reply:
done